python/mxnet/gluon/trainer.py

```diff
@@ -354,14 +366,30 @@ def allreduce_grads(self):
         self._allreduce_grads()
 
     def _allreduce_grads(self):
-        if self._kvstore:
-            for i, param in enumerate(self._params):
-                if param.grad_req != 'null':
+        # nothing to reduce
+        if not self._kvstore:
+            return
+        for i, param in enumerate(self._params):
+            if param.grad_req != 'null':
 
-                    self._kvstore.push(i, param.list_grad(), priority=-i)
-                    if not self._update_on_kvstore:
-                        self._kvstore.pull(i, param.list_grad(), priority=-i,
+                grad_list = param.list_grad()
+                # sparse gradients, call push and pull separately
+                if grad_list[0].stype != 'default':
```

Comment:
Please avoid double negative condition to improve readability.

Reply:
where is the double negative condition? There is only one negative condition, right?

Reply:
Sorry, I guess I read it wrong.